test/jasmine/tests/command_test.js

```diff
@@ -0,0 +1,641 @@
+
+            expect(result).toEqual([]);
+        });
+    });
+});
+
+describe('component bindings', function() {
+    'use strict';
+
+    var gd;
+    var mock = require('@mocks/binding.json');
+
+    beforeEach(function(done) {
+        var mockCopy = Lib.extendDeep({}, mock);
+        gd = createGraphDiv();
+
+        Plotly.plot(gd, mockCopy.data, mockCopy.layout).then(done);
+    });
+
+    afterEach(function() {
+        destroyGraphDiv(gd);
+    });
+
+    it('creates an observer', function(done) {
+        var count = 0;
+        Plots.createCommandObserver(gd, {}, [
+            { method: 'restyle', args: ['marker.color', 'red'] },
+            { method: 'restyle', args: ['marker.color', 'green'] }
+        ], function(data) {
+            count++;
+            expect(data.index).toEqual(1);
+        });
+
+        // Doesn't trigger the callback:
+        Plotly.relayout(gd, 'width', 400).then(function() {
+            // Triggers the callback:
+            return Plotly.restyle(gd, 'marker.color', 'green');
+        }).then(function() {
+            // Doesn't trigger a callback:
+            return Plotly.restyle(gd, 'marker.width', 8);
+        }).then(function() {
+            expect(count).toEqual(1);
+        }).catch(fail).then(done);
+    });
+
+    it('logs a warning if unable to create an observer', function() {
+        var warnings = 0;
+        spyOn(Lib, 'warn').and.callFake(function() {
+            warnings++;
+        });
+
+        Plots.createCommandObserver(gd, {}, [
+            { method: 'restyle', args: ['marker.color', 'red'] },
+            { method: 'restyle', args: [{'line.color': 'green', 'marker.color': 'green'}] }
+        ]);
+
+        expect(warnings).toEqual(1);
+    });
+
+    it('udpates bound components when the value changes', function(done) {
+        expect(gd.layout.sliders[0].active).toBe(0);
+
+        Plotly.restyle(gd, 'marker.color', 'blue').then(function() {
+            expect(gd.layout.sliders[0].active).toBe(4);
+        }).catch(fail).then(done);
+    });
+
+    it('udpates bound components when the computed value changes', function(done) {
+        expect(gd.layout.sliders[0].active).toBe(0);
+
+        // The default line color comes from the marker color, if specified.
+        // That is, the fact that the marker color changes is just incidental, but
+        // nonetheless is bound by value to the component.
+        Plotly.restyle(gd, 'line.color', 'blue').then(function() {
+            expect(gd.layout.sliders[0].active).toBe(4);
+        }).catch(fail).then(done);
+    });
+});
+
+describe('attaching component bindings', function() {
+    'use strict';
+    var gd;
+
+    beforeEach(function(done) {
+        gd = createGraphDiv();
+        Plotly.plot(gd, [{x: [1, 2, 3], y: [1, 2, 3]}]).then(done);
+    });
+
+    afterEach(function() {
+        destroyGraphDiv(gd);
+    });
+
+    it('attaches and updates bindings for sliders', function(done) {
```

Comment:
@etpinard See tests. It checks:

- event listeners are attached when sliders are added
- event listeners are removed when sliders are update and are no longer bindable
- changing slider options triggers recomputing the lookup table so it still works